source/Calamari.Aws/Deployment/CloudFormation/CloudFormationServiceBase.cs

```diff
@@ -0,0 +1,158 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Amazon.CloudFormation;
+using Amazon.CloudFormation.Model;
+using Amazon.Runtime;
+using Calamari.Aws.Exceptions;
+using Calamari.Aws.Integration.CloudFormation;
+using Octopus.CoreUtilities;
+using Octopus.CoreUtilities.Extensions;
+
+namespace Calamari.Aws.Deployment.CloudFormation
+{
+    public abstract class CloudFormationServiceBase
+    {
+        readonly StackEventLogger stackEventLogger;
+
+        protected readonly ILog log;
+
+        protected CloudFormationServiceBase(ILog log)
+        {
+            this.log = log;
+
+            stackEventLogger = new StackEventLogger(log);
+        }
+
+        /// <summary>
+        /// The AmazonServiceException can hold additional information that is useful to include in
+        /// the log.
+        /// </summary>
+        /// <param name="exception">The exception</param>
+        protected void LogAmazonServiceException(AmazonServiceException exception)
+        {
+            exception.GetWebExceptionMessage()
+                .Tee(message => DisplayWarning("AWS-CLOUDFORMATION-ERROR-0014", message));
```

Comment:
Please don't use the `Tee` and `Map` syntax

Reply:
@slewis74 just told me this as well. This is a direct `copy and paste` from our previous implementation. Will remove all of them in a later PR at once.